titiler/api/endpoints/mosaic.py

```diff
@@ -0,0 +1,245 @@
+"""API for MosaicJSON Dataset."""
+import asyncio
+import os
+import random
+from functools import partial
+from io import BytesIO
+from typing import Callable, Sequence
+
+import mercantile
+import morecantile
+import numpy
+from cogeo_mosaic.backends import MosaicBackend
+from cogeo_mosaic.mosaic import MosaicJSON
+from cogeo_mosaic.utils import get_footprints
+from rasterio.transform import from_bounds
+from rio_tiler.io.cogeo import tile as cogeoTiler
+from rio_tiler.profiles import img_profiles
+from rio_tiler.utils import render
+
+from titiler.api.deps import CommonTileParams, TileMatrixSetNames
+from titiler.api.endpoints.cog import cog_info, tile_response_codes
+from titiler.api.utils import postprocess
+from titiler.models.metadata import cogBounds, cogInfo
+from titiler.models.mosaic import CreateMosaicJSON, UpdateMosaicJSON
+from titiler.ressources.common import drivers
+from titiler.ressources.enums import ImageMimeTypes, ImageType, PixelSelectionMethod
+from titiler.ressources.responses import ImgResponse
+
+from fastapi import APIRouter, Depends, Path, Query
+from fastapi.routing import APIRoute
+
+from starlette.concurrency import run_in_threadpool
+from starlette.exceptions import HTTPException
+from starlette.requests import Request
+from starlette.responses import Response
+
+
+def _chunks(my_list: Sequence, chuck_size: int):
+    """Yield successive n-sized chunks from l."""
+    for i in range(0, len(my_list), chuck_size):
+        yield my_list[i : i + chuck_size]
+
+
+class MosaicJSONRouter(APIRoute):
+    """Custom router to temporarily forbid usage of cogeo-mosaic STAC backend"""
+
+    def get_route_handler(self) -> Callable:
+        """Override base method (https://fastapi.tiangolo.com/advanced/custom-request-and-route/)"""
+        original_route_handler = super().get_route_handler()
+
+        async def forbid_stac_backend(request: Request) -> Response:
+            url = request.query_params.get("url")
+            if not url:
+                body = await request.json()
+                url = body.get("url")
+            if url.startswith("stac"):
+                # Raise as a validation error
+                raise HTTPException(
+                    status_code=422,
+                    detail="STAC is not currently supported for mosaicjson endpoints",
+                )
+            return await original_route_handler(request)
+
+        return forbid_stac_backend
```

Comment:
I'm not really sure we need this, the app will just fail if you try `stac...` url

Reply:
I wanted to be explicit here for now (rather than raise a `500` because we aren't handling **kwargs in `MosaicBackend`).  Besides that i agree with you.  Ideally STACBackend is fully supported in the near future for both COG/MosaicJSON and we can remove this completely.

Reply:
Removed the custom router because it happens before dependency injection which messes with parameter validation.  I think it is fine for now to throw `500` for stac backend.

https://github.com/developmentseed/titiler/pull/40/commits/180a3d1be78e1bf325c22baf5894463c252ef221